main.go

```diff
@@ -21,9 +24,33 @@ func main() {
 	}
 }
 
+func loadTLSCert(l *slog.Logger, certPath, keyPath string) func(*tls.Config) {
+	getCertificate := func(_ *tls.ClientHelloInfo) (*tls.Certificate, error) {
+		cert, err := tls.LoadX509KeyPair(certPath, keyPath)
+		if err != nil {
+            l.Error("Unable to load TLS certificates", "error", err)
+            return nil, fmt.Errorf("Unable to load TLS certificates: %w", err)
+		}
+
+		return &cert, err
+	}
+
+	return func(config *tls.Config) {
+		config.GetCertificate = getCertificate
+	}
+}
+
 func run(l *slog.Logger) error {
 	log.SetLogger(logr.FromSlogHandler(l.Handler()))
 
+	var enableTLS bool
+	var tlsCertificatePath string
+	var tlsCertificateKeyPath string
+	flag.BoolVar(&enableTLS, "enable-tls", true, "Toggle TLS enablement.")
+	flag.StringVar(&tlsCertificatePath, "cert-path", "", "Path to TLS certificate store.")
+	flag.StringVar(&tlsCertificateKeyPath, "key-path", "", "Path to TLS private key.")
```

Comment:
As `enable-tls` defaults to true, do we need also these two to have defaults?

Reply:
What would the default be?  I don't know if there's a standard or conventional location to put TLS keys; I've seen `/etc/ssl/`, `/etc/tls/`, `/mnt/`, `/etc/webhook/` (though that seems to be a webhook-specific convention).